projects/sandbox/data_generation/data_generation/generate_background.py

```diff
@@ -88,33 +99,62 @@ def main(
         "from {} to {}".format(*segment)
     )
 
-    with h5py.File(datadir / "background.h5", "w") as f:
-        for ifo in ifos:
+    background_data = {}
+    for ifo in ifos:
 
-            # find frame files
-            files = find_urls(
-                site=ifo.strip("1"),
-                frametype=f"{ifo}_{frame_type}",
-                gpsstart=start,
-                gpsend=stop,
-                urltype="file",
-            )
-            data = TimeSeries.read(
-                files,
-                channel=f"{ifo}:{channel}",
-                start=segment[0],
-                end=segment[1],
+        # find frame files
+        files = find_urls(
+            site=ifo.strip("1"),
+            frametype=f"{ifo}_{frame_type}",
+            gpsstart=start,
+            gpsend=stop,
+            urltype="file",
+        )
+        data = TimeSeries.read(
+            files,
+            channel=f"{ifo}:{channel}",
+            start=segment[0],
+            end=segment[1],
+        )
+
+        # resample
+        data = data.resample(sample_rate)
+
+        if np.isnan(data).any():
+            raise ValueError(
+                f"The background for ifo {ifo} contains NaN values"
             )
 
-            # resample
-            data = data.resample(sample_rate)
+        if gaussian:
+            logging.info(f"Generating gaussian noise from psd for ifo {ifo}")
+
+            # TODO: what should determine this?
+            df = 1 / 8.0
+            if psd_file is not None:
```

Comment:
Let's add a flag, something like `is_asd` since some of the noise in DCC are asd instead of psd.